### PR TITLE
cephinspector: try to obtain disk id of journal and data partitions

### PR DESCRIPTION
**Pull request by kmroz**

### CODE REVIEW

**Comment by kmroz** [comment]
Need to cover nvme formats... dnm for now.

**Comment by kmroz** [comment]
@tserong / @swiftgist - added the nvme check, should be good.

File: srv/salt/_modules/cephinspector.py [hunk 1]
```diff
@@ -6,17 +6,41 @@
 from subprocess import Popen, PIPE
 import json
 
+def _get_disk_id(partition):
+    """
+    Return the disk id of a partition/device, or an empty string if not available.
+    """
+    disk_id_cmd = Popen("find -L /dev/disk/by-id -samefile " + partition + " \( -name ata* -o -name nvme* \)", stdout=PIPE, stderr=PIPE, shell=True)
```
Comment (tserong) [inline srv/salt/_modules/cephinspector.py]: Do we care about disks whose names start with "scsi" (full disclosure: I don't really know how this by-id generation stuff works)

File: srv/salt/_modules/cephinspector.py [hunk 2]
```diff
@@ -6,17 +6,41 @@
...
 def _append_to_ceph_disk(ceph_disks, partition, journal_dev):
     """
     Populate ceph_disks dictionary with data and journal partitions.
     """
+
+    # We don't care about the trailing number on journal_dev.
+    journal_dev = journal_dev.rstrip("1234567890")
+    # For nvme journal, strip the trailing 'p' as well.
+    if "nvme" in journal_dev:
+        journal_dev = journal_dev[:-1]
```
Comment (tserong) [inline srv/salt/_modules/cephinspector.py]: Why don't we care about the trailing numbers and 'p'? Does ceph only care about the base device, not the specific journal partition if the device is shared between multiple OSDs?

File: srv/salt/_modules/cephinspector.py (hunk 1 above)
Comment (kmroz) [inline srv/salt/_modules/cephinspector.py]: Talked with @swiftgist about that.  Apparently we only care about ata/nvme.

File: srv/salt/_modules/cephinspector.py (hunk 2 above)
Comment (kmroz) [inline srv/salt/_modules/cephinspector.py]: As far as I understand, our (DeepSea's) storage role yml files operate on base device for data and journal paritions.

**Pull request merged**